### PR TITLE
README.md: add refrences to related projects

### PR DESCRIPTION
**Pull request by kfessel**
### Contribution description

Add related projects, advertise Ariel OS

### CODE REVIEW

File: README.md
Comment (jkarinkl) [inline README.md:56]: [Ariel OS](https://github.com/ariel-os/ariel-os) is an offspring of RIOT rewritten in Rust following RIOTs targets.

(grammar)

Comment (miri64) [reply README.md:56]: Also: "RIOTs" is plural, I think you mean posessive "RIOT's".

Comment (Teufelchen1) [reply README.md:56]: s/targets/vision? I mean, we even have a doc page for that: https://doc.riot-os.org/vision.html

File: README.md
Comment (miri64) [inline README.md:56]: > s/targets/vision? I mean, we even have a doc page for that: https://doc.riot-os.org/vision.html

Maybe this page also should be linked then ;-)

Comment (kfessel) [reply README.md:56]:
```suggestion
[Ariel OS](https://github.com/ariel-os/ariel-os) is an offspring of RIOT written in Rust following RIOT's targets and [vision](https://doc.riot-os.org/vision.html).
```

**Review by kaspar030 — APPROVED** [review]
My ACK you got.

**Comment by aabadie** [comment]
There is a typo in the [commit](f4a87a2fe77233f433268d427ce35b742dce8c7e) message (ad => add). I fixed the PR title already. :eyes:

**Comment by kfessel** [comment]
ad(vertisement)

**Comment by OlegHahm** [comment]
Compared to the rest of the README.md I'm missing a line break after 80 characters.

**Comment by OlegHahm** [comment]
Also, I wouldn't put it into the RIOT intro text but rather add an individual headline like "Related projects" or "Spin-offs".

**Comment by kfessel** [comment]
> Compared to the rest of the README.md I'm missing a line break after 80 characters.

after pushing the fix for that i realized  its not hard at 80 but gladly it is soft (-> first line is 80 but second 83 should be ok)

> Also, I wouldn't put it into the RIOT intro text but rather add an individual headline like "Related projects" or "Spin-offs".

even though i don't think the "battery included on par" part in the Ariel OS intro is a bit optimistic ( probably just different kindof batterys), the Ariel OS - Readme refers to RIOT in the intro as well

**Comment by kfessel** [comment]
somehow ariel having they own subheading far down looks strange 
https://github.com/kfessel/RIOT/blob/6a0c3184374975c446b3179bb6acf08674356e0e/README.md
vs
https://github.com/kfessel/RIOT/blob/c31a3b47e39894b19050ba02406f2aaf66040618/README.md

**Comment by maribu** [comment]
Please squash :)

**Comment by maribu** [comment]
The rust support should IMO be a separate PR

**Comment by waehlisch** [comment]
agree with Oleg. even the current proposal is misleading. the current version shouldn't be merged.

**Comment by waehlisch** [comment]
the term "offspring" doesn't fit.

it may fit in awesome-riot-os under the caption "Projects inspired by RIOT".

File: README.md
Comment (waehlisch) [inline README.md:87]: this text doesn't reflect reality. Ariel-OS is not an offspring.

a reference to Ariel is better placed in the repo https://github.com/RIOT-OS/awesome-riot-os under "Projects inspired by RIOT".

more details, if necessary, could be discussed in an FAQ entry on the RIOT webpage, similar clarification should then be added to Ariel.

**Review by waehlisch — DISMISSED** [review]

File: README.md
Comment (kaspar030) [reply README.md:87]: > Ariel-OS is not an offspring.

When would a project qualify as offspring of another?

File: README.md
Comment (maribu) [reply README.md:84]: I think just `Rust` would be misleading, as readers may assume RIOT would be written primarily/purely in Rust.

Probably it would make sense to add some more structure. E.g.

 * Platforms Supported
    - 200+ boards based on AVR, MSP430, ESP8266, ESP32, RISC-V, ARM7, and ARM Cortex-M
    - the native port allows to run RIOT as-is on Linux and BSD.
      Multiple instances of RIOT running on a single machine can also be
      interconnected via a simple virtual Ethernet bridge or via a simulated
      IEEE 802.15.4 network (ZEP)
 * Programming Environment
    - Write native code in C, C++, or Rust
    - Rely upon POSIX APIs, user friendly hardware abstractions, and standard embedded Rust interfaces
    - Integrate dynamic runtimes such as MicroPython, elm (JavaScript), and WASM
 * Network Features and Protocols
    - IPv6
    - 6LoWPAN (RFC4944, RFC6282, and RFC6775)
    - UDP
    - RPL (storing mode, P2P mode)
    - CoAP
    - MQTT
    - Bluetooth (BLE) via [NimBLE](https://github.com/apache/mynewt-nimble)
    - LoRaWAN
    - UWB
    - CNN-Lite
 * System Features
    - a preemptive, tickless scheduler with priorities
    - flexible memory management
    - high resolution, long-term timers
    - MTD abstraction layer for memory devices
 * Security Features
    - OTA updates via SUIT
    - PSA Cryptographic API
    - DTLS, EDHOC, ...
    - ...
 * Device Drivers
    - SPI, I2C, UART, USB, ...
    - Various sensors: Environmental sensors, IMUs, magnetometers, particulate matter sensors, ...
    - Various actuators: Servos, motors, dimmers, switches, RGB LEDs, ...
    - Displays and touchscreens
    - GNSS modules
    - ...

But as said, I'd rather have this discussion split out.

File: README.md
Comment (maribu) [reply README.md:87]: > this text doesn't reflect reality. Ariel-OS is not an offspring.

The term offspring here is not be understood in a literal, biological meaning but in a metaphorical meaning.

> a reference to Ariel is better placed in the repo https://github.com/RIOT-OS/awesome-riot-os under "Projects inspired by RIOT".

The "asesome-<project>" naming has a well established meaning that is rather narrow. Projects making use and resources on RIOT do fit in there, projects inspired by RIOT not.

Ariel OS is a research project started by a sizeable and highly active part of the RIOT community with a shared vision and multiple contributors active in both projects. There is a common understanding that we want to share code between the two OS when this is technically sensible. This makes the relationship between Ariel OS and RIOT special. This special relationship is also reflected by the README in the Ariel OS repo.

**Comment by maribu** [comment]
Maybe this is a tool for efficiently finding out how the RIOT community feels about adding the reference to the README.md: https://forum.riot-os.org/t/should-we-add-a-reference-to-ariel-os-in-the-riots-readme-md/4429

File: README.md
Comment (waehlisch) [reply README.md:87]: > > this text doesn't reflect reality. Ariel-OS is not an offspring.

> The term offspring here is not be understood in a literal, biological meaning but in a metaphorical meaning.

i was writing my comment under the assumption that we are not in the bio sphere, but the wording neither fits in connotation nor denotation.

> Ariel OS is a research project started by a sizeable and highly active part of the RIOT community with a shared vision and multiple contributors active in both projects.

Ariel currently counts eleven contributors, seven of them contributed to RIOT (out of these seven only three contributed more >10 commits to Ariel).  i'm not in favor of number crunching but "sizeable and highly active part of the RIOT community" might seem a rather optimistic perspective.

> This special relationship is also reflected by the README in the Ariel OS repo.

the README says "Rust & RIOT combined for ergonomic embedded development" and "Overall, Ariel OS gives you a 'batteries-included' experience, on par with [RIOT](https://github.com/RIOT-OS/RIOT)."

originally, the Ariel README also included "~~Ariel OS links with many components of [RIOT OS](https://github.com/RIOT-OS/RIOT), which is licenced under the terms of LGPLv2.1.~~" but this was removed in a recent PR (luckily, because the crossing out didn't look good ...).

Comment (bergzand) [reply README.md:87]: > i was writing my comment under the assumption that we are not in the bio sphere, but the wording neither fits in connotation nor denotation.

Given that you seem to disagree strongly with using 'offspring' here, what's required for a project to be offspring of another project? Because the rest of the participants in the discourse here seem to agree on on offspring as a good description

> Ariel currently counts eleven contributors, seven of them contributed to RIOT (out of these seven only three contributed more >10 commits to Ariel). i'm not in favor of number crunching but "sizeable and highly active part of the RIOT community" might seem a rather optimistic perspective.

I think we both can agree that basing the level of involvement of different contributers purely on number of commits gives a very skewed and imprecise view. Maybe these numbers that gives you the impression that the statement made above by @maribu is optimistic, but I don't think any counter claims about this can be made purely based on commit history or lines changed.

> originally, the Ariel README also included "~Ariel OS links with many components of [RIOT OS](https://github.com/RIOT-OS/RIOT), which is licenced under the terms of LGPLv2.1.~" but this was removed in a recent PR (luckily, because the crossing out didn't look good ...).

:+1:

Comment (waehlisch) [reply README.md:87]: > Given that you seem to disagree strongly with using 'offspring' here, what's required for a project to be offspring of another project?

an offspring finally replaces its parents. i don't think that we want to give the impression that RIOT will be replaced by Ariel-OS, neither now nor in the long-term. furthermore, when Ariel-OS becomes mature (as most children do), we would need to change the README. why not just find wording that will last from the beginning?

anyhow, the current discussion shows that the proposed text is at least misleading. if prominently highlighted text about Ariel goes into the README, i suggest to be much more clearer. similar text then should also be added to the Arial OS README.

Comment (bergzand) [reply README.md:87]: Do you have a suggestion for alternative phrasing for the readme here?

Comment (kfessel) [reply README.md:87]: I am an offspring and have no intention to replace my ancestors living or dead. I got and gave. I learned and taught. I think that my ancestors are family and we share things and help each other. My purpose is not to replace anyone but be me. 

The ideas (vison and goals) and learnings (e.g. batteries included) of RIOT are one big part of Ariel-OS the other big one is Rust with its flourishing embedded ecosystem (rather than depending on each individual manufacturer gracefully providing their own library that then has to be adapted). 

There is a huge number of developers that like to work with C or that depend on C (manufacturer libraries) and RIOT can work for them Ariel is not able to replace RIOT there.

I don't see Ariel  as a replacement for RIOT. I think we should advertise Ariel and welcome every developer that  strives to provide to either or both OSes or on time both trough rust libraries that might be used in both.

Also we do not need to change wording if Ariel matures it will never be Riots ancestor or sibling it started from RIOT.

And even if we need to change wording or placement that is what PR are for.

Comment (chrysn) [reply README.md:87]: > an offspring finally replaces its parents.

Oh my, I hope not! (Both for RIOT's sake and my own personal life).

More seriously though. I didn't dig through all the literature to reliably state "it's used that way", but as a data point, Silberschatz's Operating Systems Concepts ([as used in Operating Systems introduction courses chapter 20](https://www-users.cse.umn.edu/~kauffman/4061/syllabus.html)) uses ancestry terminology even between operating systems where "communication system and philosophy are derived from" each other. Here we have much more: Beyond the designs and philosophy, which are clearly derived from RIOT OS, the code started out as implementing the very same interfaces. It has moved on from that, because it turned out to be unsuitable to translate words instead of meaning, but there has been a shared code starting point, and there is shared code available once more.

> similar text then should also be added to the Arial OS README.

I opened a pull request on that (https://github.com/ariel-os/ariel-os/pull/587) promptly after this one set out, using the terminology Karl suggested:

> Ariel OS is an offspring of [RIOT OS](https://github.com/RIOT-OS/RIOT). It shares many of its goals and [visions](https://doc.riot-os.org/vision.html), such as portability and providing a 'batteries-included' experience.

The current text in Ariel's README is admittedly unclear, to no small part due to the very lineage situation described above; let's update our READMEs to make things better understood.

Comment (tcschmidt) [reply README.md:87]: > ancestry terminology even between operating systems where "communication system and philosophy are derived from" each other.

The problem seems: Ariel is not an **Ancester** of RIOT. It is a different project with this personal overlap of 3 people. Maybe the three feel inspired by the RIOT work, which they may want to state on the Ariel pages---such position statements belong there.

Comment (OlegHahm) [reply README.md:87]: Even under that assumption (haven't verified) that doesn't seem to contradict what @chrysn quoted from Silberschatz.

**Comment by OlegHahm** [comment]
Did you intentionally move the reference back in the intro section? (I somehow lost track of the changes and do not trust the Github UI here.)

File: README.md
Comment (waehlisch) [reply README.md:87]: i'm confused. in the text @chrysn was writing above, i don't see a clear quote. on the webpage referenced, the term "offspring" doesn't occur. as i didn't attend CSCI 4061 at the University of Minnesota, i cannot judge about the content of the course. and i currently don't have access to the 9th Edition Operating System Concepts by Silberschatz, but at least in the [10th edition](https://codex.cs.yale.edu/avi/os-book/OS10/practice-exercises/PDF-practice-solu-dir/20.pdf), the word "offspring" is not included in Chapter 20 -- maybe my pdf reader search function doesn't work but i find other words.

anyhow, it starts getting too confusing.

there are three items to clarify:

(1) should Ariel be mentioned in the README?
(2) if it is mentioned, where should the text be placed exactly in the README?
(3) if it is mentioned, what is the exact wording (spinoff vs. offspring vs. ...)?

preferably,  all three questions are clarified separately.

Comment (bergzand) [reply README.md:87]: > anyhow, it starts getting too confusing.

The main question here is about the phrasing of the exact reference to Ariel OS. The discussion mainly evolved around whether offspring is an applicable term here. No alternative suggestions have been offered so far.

> (1) should Ariel be mentioned in the README?

I count 3 approvals. Furthermore 5 participants in this conversation submitted comments that implied agreement, either through suggesting slight alternatives in the exact location or commenting on grammatical and spelling mistakes.

> (2) if it is mentioned, where should the text be placed exactly in the README?

This converged to either in the intro or in a new related work section. From what I can see nobody is strongly against a separate related work section and this was the state of this PR a [few commits back](https://github.com/RIOT-OS/RIOT/blob/6a0c3184374975c446b3179bb6acf08674356e0e/README.md#related-project)

> (3) if it is mentioned, what is the exact wording (spinoff vs. offspring vs. ...)?

I see 8 participants here that are fine with the current phrasing, and 2 against with currently no alternative suggestions. 

@waehlisch, @tcschmidt as you are the only participants in the discussion against the direction of this PR, could you either provide your answers to the the [three questions](https://github.com/RIOT-OS/RIOT/pull/21052#discussion_r1864794040) above or dismiss the blocking review.

Comment (OlegHahm) [reply README.md:87]: > I count 3 approvals. Furthermore 5 participants in this conversation submitted comments that implied agreement

And 9 positive votes in the [forum post](https://forum.riot-os.org/t/should-we-add-a-reference-to-ariel-os-in-the-riots-readme-md/4429) where 8 of them come from maintainers.

Comment (waehlisch) [reply README.md:87]: > The main question here is about the phrasing of the exact reference to Ariel OS.

debatable, given the caption and contribution description my understanding was different.

> No alternative suggestions have been offered so far.

not true. @OlegHahm was suggesting "spin-off" under the section "Related projects" and i gave it a +1 before i started commenting.

> this was the state of this PR a [few commits back](https://github.com/RIOT-OS/RIOT/blob/6a0c3184374975c446b3179bb6acf08674356e0e/README.md#related-project)

but this is not the state i was commenting on, nor the state that we currently discuss.

to make my position even more clearer: i don't think that any discussion about related projects belongs to the README of RIOT. in the RIOT README, the content in its core should be about RIOT, additional information can be reached via references. the relation between Ariel and RIOT is something that should be discussed in an FAQ (happy to see a generic reference in the README to our FAQ). if there is rough consensus that text about Ariel should be included in the README, i'm strongly against the term offspring but i don't have anything against calling Ariel a spin-off of RIOT.

Comment (waehlisch) [reply README.md:87]: > @waehlisch the 9th edition including @chrysn's quote is available online at https://github.com/gg-daddy/ebooks/blob/master/Operating%20System%20Concepts%2C%209th%20Edition.pdf, for instance.

thanks, but doesn't help me to agree on what @chrysn was writing. anyhow, i don't think that this is the right forum to discuss this further.

Comment (chrysn) [reply README.md:87]: It is the forum to discuss it because there is a 4-line PR blocked by a position that many do not comprehend.

I brought up that book because you claim in the request-for-change position that the "offspring" term is inaccurate. As I found it accurate based on having seen a lot of informal descriptions of OS lineages, I looked up the next best available courses on "what is an OS", as to see whether those genealogical terms are just used informally or also in textbooks. So no need to look up University Minnesota courses – that's just where I found that book. Chapter 20 of that book (here my reference was inaccurate: the link made it sound like it's a chapter in the course) does what I expect OS text books to do: Use terminology of ancestry rather loosely. They do not use "offspring" (AIU nobody here claimed that), but they do use the term "ancestry" even for operating systems that have less in common than RIOT and Ariel. Full relevant quote:

> The Mach operating system traces its ancestry to the Accent operating system developed at Carnegie Mellon University (CMU). Mach’s communication system and philosophy are derived from Accent, but many other significant portions of the system (for example, the virtual memory system and task and thread management) were developed from scratch.

So from where I'm standing, terminology of being a family or lineage of operating systems has been in wide use, and the references I've looked up on that so far seem to support that. You probably have more experience throwing "ancestry" and "operating systems" into the relevant publication search engines – but maybe you can also clarify whether the opposition here is on the amount of influence Ariel claims RIOT had on it, whether you disagree that the terminology is the widespread one for this kind of relations, or whether you disagree that "offspring" is evocative of the ancestry terminology.

Comment (benpicco) [inline README.md:87]: I always liked littlefs' [related projects](https://github.com/littlefs-project/littlefs?tab=readme-ov-file#related-projects) section. I think it's a sign of humility to know that you are not the only game in town, but that you aspire to a friendly competition instead claiming that you are the greatest and nothing can compare.

It also helps with understanding what the project is about. "What is RIOT? Ah, it's like Zephyr or Contiki, so not something I'd run as the main OS on my laptop or in a container in the cloud."

```suggestion
### Related projects

- [Ariel OS](https://github.com/ariel-os/ariel-os) is an offspring of RIOT written
in Rust following RIOT's targets and [vision](https://doc.riot-os.org/vision.html).

- [Contiki](https://github.com/contiki-ng/contiki-ng) was an inspiration for starting RIOT, a operating system for constrained devices with a focus on networking.

- [NuttX](https://github.com/apache/nuttx) is another general purpose microcontroller OS with a focus on bringing POSIX to MCUs

- [RT-Thread](https://github.com/RT-Thread/rt-thread) is a microcontroller operating system with strong roots in China and a large community there, it has extensive support for MCUs from Chinese vendors, but also for western ones.

- [Zephyr](https://github.com/zephyrproject-rtos/zephyr) is general-purpose operating system for microcontrollers, shepparded by the Linux foundation. 

```

phrasing is of course up for suggestions, e.g. I know only very little about NuttX and RT-Thread.

My criteria for inclusion would be that it is a 'batteries included' OS with a proper networking stack.

Comment (waehlisch) [reply README.md:87]: > I always liked littlefs' [related projects](https://github.com/littlefs-project/littlefs?tab=readme-ov-file#related-projects) section. I think it's a sign of humility to know that you are not the only game in town, but that you aspire to a friendly competition instead claiming that you are the greatest and nothing can compare.
> 
> It also helps with understanding what the project is about. "What is RIOT? Ah, it's like Zephyr or Contiki, so not something I'd run as the main OS on my laptop or in a container in the cloud."
> 
> phrasing is of course up for suggestions, e.g. I know only very little about NuttX and RT-Thread.
> 
> My criteria for inclusion would be that it is a 'batteries included' OS with a proper networking stack.

sounds like a decent suggestion. i sill would replace "offspring" by "spin-off". i also like the caption because "related" is a better fit than "alternative" (alternative really depends on the context).

Comment (maribu) [reply README.md:87]: Let's go bike-shedding: I'd move Zephyr up, as I believe it to be the most obvious alternative to RIOT, one below Ariel OS.

We use [title case](https://titlecaseconverter.com/) for the other section headers, so " Related projects" --> "Related Projects".

**Comment by kaspar030** [comment]
Whatever the wording will converge to in the end, the friendlyness, support and solidary towards Ariel OS as shown by most of the community is heartening and inspiring. Thanks for that!! :heart:

**Comment by emmanuelsearch** [comment]
I hate to crash the party, but I find it ironic that we unable to make a clearer distinction between Zephyr (clearly a merciless opponent on all accounts) and Ariel OS, which is a brother project in fact, with whom we share core maintainers, code, infrastructure, and values).

To me this is yet another symptom of a fundamental problem we have, which has blinded and plagued RIOT for way too long.

**Comment by miri64** [comment]
> and to be concrete on what I mean by "fundamental issue": we do not have a governance.md, and we probably have been needing one since quite a long time. While that discussion belongs in its own thread, I think the issues we have in this thread are typical, and symptoms of the absence of governance.

Poor @kfessel. Now this got bikeshedded into a direction that has nothing to do with advertising Ariel OS anymore. If no one else wants to start a governance.md, I can prepare something.

**Comment by kaspar030** [comment]
> Now this got bikeshedded into a direction that has nothing to do with advertising Ariel OS anymore.

+1 for moving `GOVERNMENT.md` creation into a spin-off PR.

**Review by aabadie — COMMENTED** [review]
The PR title doesn't reflect anymore the goal of the PR (which is add related projects section).
Also there are typos in the commit messages:
1. I insist that "ad Ariel OS" is grammatically wrong and it should be "advertize Ariel OS" (American English form)
2. There's an extra "f " at the beginning of the second commit. Please also add a verb. "Add" works here. "README.md: add Related projects section".

**Comment by kfessel** [comment]
@aabadie: i use f to mark commit i intend to squash / for fixups (spare me off some static noise)

I will squash this PR into 

```
README.md: add Related Projects

advertise Ariel OS
```

this evening

File: README.md
Comment (aabadie) [inline README.md:89]:
```suggestion
- [Contiki(-NG)](https://github.com/contiki-ng/contiki-ng) was an inspiration for starting RIOT, an operating system for constrained devices with a focus on networking.
```

**Comment by miri64** [comment]
> Poor @kfessel. Now this got bikeshedded into a direction that has nothing to do with advertising Ariel OS anymore. If no one else wants to start a governance.md, I can prepare something.

https://forum.riot-os.org/t/starting-a-governance-md-document/4431 (for now maintainers and moderators only, but maybe we can involve @jkarinkl into the discussion somehow).

File: README.md
Comment (chrysn) [inline README.md:92]: just spelling:

```suggestion
- [Zephyr](https://github.com/zephyrproject-rtos/zephyr) is a general-purpose operating system for microcontrollers, shepherded by the Linux foundation.
```

**Comment by OlegHahm** [comment]
If we really want to put a link to things like Zephyr in the README we should IMO also point out what we consider as the major differences (licensing, community model, memory efficiency?) but that will probably blowup the README quite a bit and I'm not sure we really want to do this. I'm fine with citing other (mostly unrelated) projects but I don't think that a README is the best place to do that.

File: README.md
Comment (maribu) [inline README.md:88]:
```suggestion
- [**Ariel OS**](https://github.com/ariel-os/ariel-os) is young IoT OS with a
  focus on security and memory safety that is inspired by RIOT and follows the
  same goals and [vision](https://doc.riot-os.org/vision.html).
```

How about this wording It avoids the offspring/spinoff wording, that seem to be the main point of friction.

**Comment by kaspar030** [comment]
> i sill would replace "offspring" by "spin-off".

Contrary to offspring, spin-off is actually well defined around academia and industry. It means a by-product, an accidental development of something else internal at the project. `murdock(-ng)`, `philipp`, `laze`, `cosy` are spin-offs, side projects doing something different that have been spun off the main project. As Ariel OS was already outside the RIOT organisation before development started, the term spin-off is not accurate here.

**Comment by tcschmidt** [comment]
> > i sill would replace "offspring" by "spin-off".
> 
> Contrary to offspring, spin-off is actually well defined around academia and industry. It means a by-product, an accidental development of something else internal at the project. 

'accidental' is certainly not correct, e.g., Infineon is a spin-off of Siemens: https://en.wikipedia.org/wiki/Infineon_Technologies

> As Ariel OS was already outside the RIOT organisation before development started, the term spin-off is not accurate here.

Agree. This underlines that Ariel is a project independent of RIOT. Sharing code and ideas is the fundamental concept of Open Source - not specific to Ariel nor special to RIOT. IMO, this boils the discussion down (again) to 'inspired by' - Ariel developers may of course state 'Ariel was inspired by RIOT', if they wish to.

**Comment by maribu** [comment]
> IMO, this boils the discussion down (again) to 'inspired by' - Ariel developers may of course state 'Ariel was inspired by RIOT', if they wish to.

They can and the actually do. Not in the exact wording. As you seem to have a number of suggestions on improving Ariel OS's README: They are open for PRs. Any discussion on Ariel OS's README is much better done there, as there this discussion can actually be acted upon.

The point of this PR is to add a reference to Ariel OS in the README. The arguments against this have been repeated a few times now, with little to no variant in the reasoning. That does not have seemed to convinced anyone, while at the same time more and more support for adding the reference has been voiced, e.g. in [here](https://forum.riot-os.org/t/should-we-add-a-reference-to-ariel-os-in-the-riots-readme-md/4429).

I think it is time to call this a rough consensus and move on.

**Comment by maribu** [comment]
> IMO also point out what we consider as the major differences (licensing, community model, memory efficiency?)

We do have that [in the FAQ](https://www.riot-os.org/#faq-answer-12). We could refer to that in the README.md for those interested in the detailed look. I think adding a small reference does help users who are new to RIOT but know one or more of the references to get what the target group of RIOT is. IMO an in-detail discussion is too long for a README, but adding a ref to such would be IMO quite helpful.

**Comment by maribu** [comment]
I count three ACKs and overwhelming support in the forum. Let's merge this and refine from there

**Comment by maribu** [comment]
First refinement PR is open: https://github.com/RIOT-OS/RIOT/pull/21061

This adds a reference to the in-detail comparison between RIOT and other OSes.

**Comment by waehlisch** [comment]
@maribu, this merge was too fast. we had somehow a consensus on mentioning related projects, but we didn't have any consensus on the wording regarding Ariel. even you suggested an alternative. just merging is really disappointing.

**Comment by OlegHahm** [comment]
I have to agree and disagree with @waehlisch here: this was merged to fast but in my opinion we haven't had consensus on mentioning other related projects.
This PR started to add the reference to Ariel OS that was concluded at the summit. And that's what have been acked by three maintainers. After these acks the content of the PR underwent a major change and at least I was not happy with that (as stated in https://github.com/RIOT-OS/RIOT/pull/21052#issuecomment-2511820836). This is IMO not how a PR review should work. Plus: there was no need to rush it.

**Comment by maribu** [comment]
> @maribu, this merge was too fast.

We regularly have more complex PRs with a lot less ACKs and discussions merged faster. Would you mind to elaborate on why you think this was too fast? 

That said, if you have ideas on how to improve, PRs are always welcome.

**Comment by miri64** [comment]
> We regularly have more complex PRs with a lot less ACKs and discussions merged faster.

Especially when it comes to code PRs (yes, this is “just” a doc PR), this sounds more like a bad symptom than an argument... I share @OlegHahm's analysis how the ACKs went. None where after the latest set of changes.

Additionally, there were complaints in the past [few days!] of PRs being ACK'd by mentors/advisors/colleagues and disregarding if I like the content of this PR or how the discussion went or not, there is a certain bad look to it, that 2/3 ACKs came from colleagues of the author and the third ACK from one of the founder/leaders of Ariel OS.

**Comment by tcschmidt** [comment]
> > @maribu, this merge was too fast.
> 
> We regularly have more complex PRs with a lot less ACKs and discussions merged faster. Would you mind to elaborate on why you think this was too fast?
> 
I don't think so: decorating the front readme with a (IMO completely misplaced) random selection of related projects does not look good. In other words: This operation has been much more complex than apparently perceived.

**Comment by emmanuelsearch** [comment]
> there were complaints in the past [few days!] of PRs being ACK'd by mentors/advisors/colleagues and disregarding if I like the content of this PR or how the discussion went or not, there is a certain bad look to it, that 2/3 ACKs came from colleagues of the author and the third ACK from one of the founder/leaders of Ariel OS.

I agree it's reasonable to suspect (and fight against) collusion at work in some factions within RIOT.
*However* this is likely *not* the case here.
(This company has not vested interest in Ariel whatsoever, and this PR was only touching a README, not a controversial functionality).
Hence, I don't see the point of bringing this up here.
The problem is probably elsewhere?

**Pull request merged**
